viz/LaserLine.cpp

```diff
@@ -2,44 +2,150 @@
 #include "LaserLine.hpp"
 
 using namespace vizkit3d;
+using namespace vizkit3d_laser_line;
 
-struct LaserLine::Data {
-    // Copy of the value given to updateDataIntern.
-    //
-    // Making a copy is required because of how OSG works
-    base::samples::LaserScan data;
-};
 
+#define milimeters_to_meters(x) (x) / 1000.0
 
 LaserLine::LaserLine()
-    : p(new Data)
 {
+    color = osg::Vec4(0.0, 1.0, 0.0, 0.0);
```

Comment:
OK. Much better C++ style is to use initialization lists

~~~
LaserLine::LaserLine()
   : color(osg::Vec4(0.0, 1.0, 0.0, 0.0))
   , vertices(new osg::Vec3Array())
   , ...
~~~

The reason is that it handles destruction properly if any of the initializations fails.